### PR TITLE
Add debug symbols upload settings

### LINKED ISSUE #84: Automatic Symbol Upload Fails due to missing properties file
### Environment

How do you use Sentry?
Sentry.io

Which version of the SDK?

Downloaded Git source code and integrated manually (And copied dependencies from package job)

Which version of Unreal?

5.0.0 Build on TeamCity of the Client 

### Steps to Reproduce

1. Submitted Sentry plugin with automatic upload enabled
2. Built the client (And cooked) via TeamCity

### Expected Result
Symbols automatically uploaded 

### Actual Result

[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] Executing post build script (PostBuild-1.bat) (0:02.53 at +1:40)
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] Sentry: Start debug symbols upload
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] Sentry: Parse project settings
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] Sentry: Upload started using PropertiesFile 'E:/TCWorkDir/213dfa932477ea94/Project/'
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] error: Failed to load file referenced by SENTRY_PROPERTIES (E:/TCWorkDir/213dfa932477ea94/Project/)
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   caused by: Access is denied. (os error 5)
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] thread 'main' panicked at 'Config not bound yet': src\config.rs:85
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] 
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    0: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    1: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    2: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    3: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    4: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    5: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    6: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    7: git_midx_writer_new
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    8: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]    9: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   10: git_midx_writer_new
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   11: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   12: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   13: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   14: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   15: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   16: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   17: git_libgit2_features
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   18: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   19: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64]   20: <unknown>
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] 
[15:37:39] :		 [[5/8] Compile ProjectPlayFabClient Win64] Sentry: Upload finished

### LINKED ISSUE #93: Automatic properties file generation for debug symbols upload
Currently there is no way to generate `sentry.properties` file required for automatic debug symbols upload but creating it manually in a specific location. The solution is to extend the plugin settings editor menu in a way that allows entering project name, organization slug and authentication token conveniently and without the risk of compromising sensitive user data (meaning those won't be stored in `DefaultEngine.ini`).

Also, this should resolve #84

### PR DESCRIPTION
**Pull request by tustanivsky**
Added custom editor menu that allows generating `sentry.properties` file in project root directory automatically without storing user credentials in `DefaultEngine.ini` that is under source control. Properties file is updated whenever the user edits project/organization/token fields.

SentryPlayground_-_Unreal_Editor

Closes #93 
Closes #84

### CODE REVIEW

File: plugin-dev/Scripts/upload-debug-symbols.sh [hunk 1]
```diff
@@ -38,9 +38,9 @@ if [ $UPLOAD_SYMBOLS != "True" ]; then
     exit
 fi
 
-PROP_FILE_PATH=$(awk -F "=" '/PropertiesFilePath/ {print $2}' ${CONFIG_PATH}/DefaultEngine.ini)
+#PROP_FILE_PATH=$(awk -F "=" '/PropertiesFilePath/ {print $2}' ${CONFIG_PATH}/DefaultEngine.ini)
```
Comment (vaind) [inline plugin-dev/Scripts/upload-debug-symbols.sh:41]: nit: remove old code?

File: plugin-dev/Scripts/upload-debug-symbols.sh [hunk 2]
```diff
@@ -38,9 +38,9 @@ if [ $UPLOAD_SYMBOLS != "True" ]; then
...
 
-export SENTRY_PROPERTIES="$projectPath/$PROP_FILE_PATH"
+export SENTRY_PROPERTIES="$projectPath/sentry.properties"
```
Comment (vaind) [inline plugin-dev/Scripts/upload-debug-symbols.sh:43]: maybe add the same check & warning that the file doesn't exist as on Windows?

**Review by vaind — CHANGES_REQUESTED** [review]

File: plugin-dev/Source/Sentry/Private/SentrySettings.cpp [hunk 3]
```diff
@@ -2,10 +2,30 @@
...
+{
+	DsnUrl = TEXT("");
+	Release = TEXT("");
+
+	LoadDebugSymbolsProperties();
+}
+
+void USentrySettings::LoadDebugSymbolsProperties()
+{
+	const FString PropertiesFilePath = FPaths::Combine(FPaths::ProjectDir(), TEXT("sentry.properties"));
```
[the remark this answers is not in the document]
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Private/SentrySettings.cpp:20]: I believe it's fine to put `sentry.properties` alongside with `.uproject` file while `Config` folder is meant for UE-specific configuration files. There are no strict rules regarding this matter though

File: plugin-dev/Scripts/upload-debug-symbols.sh (hunk 2 above)
Comment (tustanivsky) [reply plugin-dev/Scripts/upload-debug-symbols.sh:46]: Sure, that's a good point - added a corresponding check

File: plugin-dev/Scripts/upload-debug-symbols-win.ps1 [hunk 4]
```diff
@@ -77,9 +77,13 @@ If ("$UploadSymbols".ToLower() -ne "true")
...
 
-Write-Host "Sentry: Parse project settings"
+$PropertiesFile = "$ProjectPath/sentry.properties"
 
-$PropertiesFile = "$ProjectPath/$($ConfigIni.$SentrySettingsSection.PropertiesFilePath)".Replace('\', '/')
+If (-not (Test-Path -Path $PropertiesFile -PathType Leaf))
+{
+    Write-Host "Sentry: Automatic symbols upload is not configured in plugin settings or properties file is missing. Skipping..."
+    Exit
+}
```
[the remark this answers is not in the document]
Comment (tustanivsky) [reply plugin-dev/Scripts/upload-debug-symbols-win.ps1:86]: Agree - updated this part a bit

File: plugin-dev/Scripts/upload-debug-symbols-win.ps1 [hunk 5]
```diff
@@ -77,9 +77,13 @@ If ("$UploadSymbols".ToLower() -ne "true")
     Exit
 }
 
-Write-Host "Sentry: Parse project settings"
+$PropertiesFile = "$ProjectPath/sentry.properties"
 
-$PropertiesFile = "$ProjectPath/$($ConfigIni.$SentrySettingsSection.PropertiesFilePath)".Replace('\', '/')
+If (-not (Test-Path -Path $PropertiesFile -PathType Leaf))
+{
+    Write-Error "Sentry: Properties file is missing"
```
Comment (vaind) [inline plugin-dev/Scripts/upload-debug-symbols-win.ps1:84]:
```suggestion
    Write-Error "Sentry: Properties file is missing: '$PropertiesFile'"
```

File: plugin-dev/Scripts/upload-debug-symbols.sh [hunk 6]
```diff
@@ -38,9 +38,12 @@ if [ $UPLOAD_SYMBOLS != "True" ]; then
     exit
 fi
 
-PROP_FILE_PATH=$(awk -F "=" '/PropertiesFilePath/ {print $2}' ${CONFIG_PATH}/DefaultEngine.ini)
+if [ ! -f "$projectPath/sentry.properties" ]; then
+    echo "Sentry: Properties file is missing"
+    exit 1
+fi
```
Comment (vaind) [inline plugin-dev/Scripts/upload-debug-symbols.sh:44]:
```suggestion
export SENTRY_PROPERTIES="$projectPath/sentry.properties"
if [ ! -f "$SENTRY_PROPERTIES" ]; then
    echo "Sentry: Properties file is missing: '$SENTRY_PROPERTIES'"
    exit 1
fi
```

File: plugin-dev/Scripts/upload-debug-symbols.sh (hunk 2 above)
Comment (vaind) [reply plugin-dev/Scripts/upload-debug-symbols.sh:47]:
```suggestion
```

**Review by vaind — APPROVED** [review]
LGTM after the additional info in the scripts (see my recent suggestions)

**Comment by vaind** [comment]
actually, maybe a changelog entry here would be useful

**Pull request merged**